packages/material/src/complex/MaterialObjectRenderer.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+    ControlElement,
+    ControlProps,
+    isObjectControl,
+    mapStateToControlProps,
+    RankedTester,
+    rankWith,
+    resolveSchema,
+    UISchemaElement
+} from '@jsonforms/core';
+import { connectToJsonForms } from '@jsonforms/react';
+import { MaterialLayoutRenderer } from '../util/layout';
+import { Card, CardContent, CardHeader } from '@material-ui/core';
+
+const createControl = (schema, path: string): UISchemaElement[] => {
+    const obj =  resolveSchema(schema, path);
+    return Object.keys(obj).map(key => {
+        return {
+            label: key,
+            type: 'Control',
+            scope: `${path}/${key}`
+        };
+    });
+};
+
+class MaterialObjectRenderer extends React.Component<ControlProps, any> {
+
+    render() {
+        const {
+            uischema,
+            schema,
+            visible,
+           } = this.props;
+
+        const controlElement = uischema as ControlElement;
+        const scope = controlElement.scope;
+        const elements: UISchemaElement[] = createControl(schema, (`${scope || '#'}/properties`));
+        const style: {[x: string]: any} = { marginBottom: '10px' };
+        if (!visible) {
+            style.display = 'none';
+        }
+
+        return (
+            <Card style={style}>
+                {!_.isEmpty(controlElement.label) && <CardHeader title={controlElement.label}/>}
+                <CardContent>
+                    <MaterialLayoutRenderer
```

Comment:
why not delegate to jsonforms similar to the allof renderer?

Reply:
Usage of the `JsonForms` component is not applicable here, since we have to deal with an array of UI schema elements.

Reply:
why can't we simply generate a uischema or use the uischemaservice as we do in the tree?

Reply:
What do you have in mind beside the usage of the `createControlElement` function?

Reply:
I could imagine that you would want to provide a full uischema and not let it generate.
I don't really see the difference here between the detail view of the tree and the inline view of the object renderer